lib/policies/oauth2-introspect/oauth2-introspect.js

```diff
@@ -0,0 +1,42 @@
+const tokenIntrospectionGenerator = require('./token-introspector');
+const BearerStrategy = require('passport-http-bearer');
+const passport = require('passport');
+
+module.exports = function (actionParams) {
+  actionParams.session = false;
+  const scannedTokens = [];
+  const tokenIntrospection = tokenIntrospectionGenerator(actionParams);
+
+  passport.use('bearer-introspect', new BearerStrategy({ passReqToCallback: true }, (req, accessToken, done) => {
+    const requestedScopes = req.egContext.apiEndpoint.scopes;
+
+    const scopeCheck = (tokenData, done) => {
+      const avaiableScopes = tokenData.scopes ? tokenData.scopes.split(' ') : [];
+
+      if (!requestedScopes.some(requestedScope => {
```

Comment:
Are you trying to check this condition `requestedScopes.every(avaiableScopes.includes)` ?

`requsted [r, w] with available [r,w,admin]` -> ok because r and w both included in `available` 

`requsted [r, w] with available [r,admin]` -> not ok because w is missing in `available`

Reply:
Yeah, that's the intention. Let me try your code.